jenkins/job-configs/kubernetes-jenkins/kubernetes-soak.yaml

```diff
@@ -109,6 +109,40 @@
                 export FAIL_ON_GCP_RESOURCE_LEAK="true"
                 export CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS="1"
                 export KUBE_NODE_OS_DISTRIBUTION="debian"
+        - 'gce-federation':
+            deploy-description: |
+                Deploy Kubernetes and federation to soak cluster using the latest successful
+                Kubernetes build every week.<br>
+                If a kubernetes-soak-continuous-e2e-gce-federation build is running,
+                this deployment build will be blocked and remain in the queue
+                until the test run is complete.<br>
+            e2e-description: |
+                Assumes Kubernetes and federation soak cluster is already deployed.<br>
+                If a kubernetes-soak-weekly-deploy-gce-federation build is enqueued,
+                builds will be blocked and remain in the queue until the
+                deployment is complete.<br>
+            branch: 'master'
+            job-env: |
+                export PROJECT="k8s-jkns-gce-federation-soak"
+            # Need the 8 essential kube-system pods ready before declaring cluster ready
+            # etcd-server, kube-apiserver, kube-controller-manager, kube-dns
+            # kube-scheduler, l7-default-backend, l7-lb-controller, kube-addon-manager
+            provider-env: |
```

Comment:
I don't fully understand the distinction between provider-env and job-env. @ixdy @spxtr might know. Perhaps @quinton-hoole might know too.

What's the difference between these values? We want to supply `FEDERATION=true` and `FEDERATION_PUSH_REPO_BASE` to both build/deploy and test jobs, and everything else only to the test job. Also, it doesn't hurt to supply all the values to both jobs. Is this config sufficient?